core/macros/src/lib.rs

```diff
@@ -62,10 +62,44 @@ pub fn enum_trait_object(args: TokenStream, item: TokenStream) -> TokenStream {
     // to the underlying type.
     let trait_methods: Vec<_> = input_trait
         .items
-        .iter()
-        .map(|item| match item {
-            TraitItem::Fn(method) => {
+        .iter_mut()
+        .filter_map(|item| match item {
+            TraitItem::Fn(ref mut method) => {
                 let method_name = &method.sig.ident;
+
+                let mut is_no_dynamic = false;
+
+                let new_attrs = method
+                    .attrs
+                    .iter()
+                    .filter(|attr| match &attr.meta {
+                        Meta::Path(path) => {
+                            if let Some(ident) = path.get_ident() {
+                                if ident == "no_dynamic" {
+                                    is_no_dynamic = true;
+
+                                    // Remove the #[no_dynamic] attribute from the
+                                    // list of method attributes.
+                                    false
+                                } else {
+                                    true
+                                }
+                            } else {
+                                true
+                            }
+                        }
+                        _ => true,
+                    })
+                    .map(|o| o.clone())
+                    .collect::<Vec<_>>();
```

Comment:
instead of copying and reassigning, can we call `method.attrs.retain(|attr| ...)`?